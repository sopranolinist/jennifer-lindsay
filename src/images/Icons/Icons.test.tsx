import { render } from '@testing-library/react';
import React from 'react';
import {
  AccountCircleIcon,
  CancelIcon,
  CloseIcon,
  DescriptionIcon,
  FirstPageIcon,
  KeyboardArrowDownIcon,
  KeyboardArrowLeftIcon,
  KeyboardArrowRightIcon,
  KeyboardArrowUpIcon,
  LastPageIcon,
  LocalLibraryIcon,
  LocalParkingIcon,
  MenuIcon,
  MoreVertIcon,
  OpenInBrowserIcon,
  OpenInNewIcon,
  PermIdentityIcon,
  PollIcon,
  SearchIcon,
  StopIcon,
  WhatsHotIcon
} from './';

describe('Icons', () => {
  test('renders the AccountCircleIcon', async done => {
    const { container } = render(<AccountCircleIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the CancelIcon', async done => {
    const { container } = render(<CancelIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the CloseIcon', async done => {
    const { container } = render(<CloseIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the KeyboardArrowDownIcon', async done => {
    const { container } = render(<KeyboardArrowDownIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the KeyboardArrowLeftIcon', async done => {
    const { container } = render(<KeyboardArrowLeftIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the KeyboardArrowRightIcon', async done => {
    const { container } = render(<KeyboardArrowRightIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the KeyboardArrowUpIcon', async done => {
    const { container } = render(<KeyboardArrowUpIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the MenuIcon', async done => {
    const { container } = render(<MenuIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the MoreVertIcon', async done => {
    const { container } = render(<MoreVertIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the PermIdentityIcon', async done => {
    const { container } = render(<PermIdentityIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the PollIcon', async done => {
    const { container } = render(<PollIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the SearchIcon', async done => {
    const { container } = render(<SearchIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the OpenInBrowserIcon', async done => {
    const { container } = render(<OpenInBrowserIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the OpenInNewIcon', async done => {
    const { container } = render(<OpenInNewIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the DescriptionIcon', async done => {
    const { container } = render(<DescriptionIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the LocalLibraryIcon', async done => {
    const { container } = render(<LocalLibraryIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the LocalParkingIcon', async done => {
    const { container } = render(<LocalParkingIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the WhatsHotIcon', async done => {
    const { container } = render(<WhatsHotIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the StopIcon', async done => {
    const { container } = render(<StopIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the FirstPageIcon', async done => {
    const { container } = render(<FirstPageIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
  test('renders the LastPageIcon', async done => {
    const { container } = render(<LastPageIcon />);
    expect(container).toMatchSnapshot();
    done();
  });
});
