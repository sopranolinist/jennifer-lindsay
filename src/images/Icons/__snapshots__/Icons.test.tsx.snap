// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders the AccountCircleIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
    />
  </svg>
</div>
`;

exports[`Icons renders the CancelIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
    />
  </svg>
</div>
`;

exports[`Icons renders the CloseIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</div>
`;

exports[`Icons renders the DescriptionIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
    />
  </svg>
</div>
`;

exports[`Icons renders the FirstPageIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
    />
  </svg>
</div>
`;

exports[`Icons renders the KeyboardArrowDownIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
    />
  </svg>
</div>
`;

exports[`Icons renders the KeyboardArrowLeftIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
    />
  </svg>
</div>
`;

exports[`Icons renders the KeyboardArrowRightIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
    />
  </svg>
</div>
`;

exports[`Icons renders the KeyboardArrowUpIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
    />
  </svg>
</div>
`;

exports[`Icons renders the LastPageIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
    />
  </svg>
</div>
`;

exports[`Icons renders the LocalLibraryIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
    />
  </svg>
</div>
`;

exports[`Icons renders the LocalParkingIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M13 3H6v18h4v-6h3c3.31 0 6-2.69 6-6s-2.69-6-6-6zm.2 8H10V7h3.2c1.1 0 2 .9 2 2s-.9 2-2 2z"
    />
  </svg>
</div>
`;

exports[`Icons renders the MenuIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</div>
`;

exports[`Icons renders the MoreVertIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
</div>
`;

exports[`Icons renders the OpenInBrowserIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 4H5c-1.11 0-2 .9-2 2v12c0 1.1.89 2 2 2h4v-2H5V8h14v10h-4v2h4c1.1 0 2-.9 2-2V6c0-1.1-.89-2-2-2zm-7 6l-4 4h3v6h2v-6h3l-4-4z"
    />
  </svg>
</div>
`;

exports[`Icons renders the OpenInNewIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
    />
  </svg>
</div>
`;

exports[`Icons renders the PermIdentityIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;

exports[`Icons renders the PollIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
    />
  </svg>
</div>
`;

exports[`Icons renders the SearchIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
    />
  </svg>
</div>
`;

exports[`Icons renders the StopIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 6h12v12H6z"
    />
  </svg>
</div>
`;

exports[`Icons renders the WhatsHotIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
    />
  </svg>
</div>
`;
